assign-doi-datacite/src/test/java/no/unit/nva/datacite/mdsclient/DataCiteMdsConnectionTest.java

```diff
@@ -0,0 +1,146 @@
+package no.unit.nva.datacite.mdsclient;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import nva.commons.utils.IoUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class DataCiteMdsConnectionTest {
+
+    public static final String DATACITE_MDS_OK_RESPONSE = "/dataciteMdsOkResponse.txt";
+    public static final String DATACITE_MDS_POST_METADATA_RESPONSE = "/dataciteMdsPostMetadataResponse.txt";
+    public static final String DATACITE_MDS_GET_DOI_RESPONSE = "/dataciteMdsGetDoiResponse.txt";
+    public static final String DATACITE_XML_RESOURCE_EXAMPLE = "/dataciteXmlResourceExample.xml";
+
+    public static final String MOCK_HOST = "nva-mock.unit.no";
+    public static final String MOCK_LANDING_PAGE_URL = "https://nva-mock.unit.no/123456789";
+    public static final String MOCK_DOI_PREFIX = "prefix";
+    public static final String MOCK_DOI = "prefix/suffix";
+    public static final String MOCK_DATACITE_XML = "mock-xml";
+    public static final String MOCK_USER = "MOCK_USER";
+    public static final String MOCK_PASSWORD = "MOCK_PASSWORD";
+    private static final int MOCK_PORT = 8888;
+
+    @Mock
+    HttpClient httpClient;
+
+    @Mock
+    HttpResponse<String> httpResponse;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testDataCiteMdsConnection() {
+        DataCiteMdsConnection dataCiteMdsConnection = new DataCiteMdsConnection(httpClient, MOCK_HOST, MOCK_PORT);
+        assertNotNull(dataCiteMdsConnection);
```

Comment:
could this ever be non null? I do not think a constructor ever returns null. Also, test name is non descriptive.